File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
        // Step 1: Merge both arrays
        vector<int> merged;
        // merged.reserve(nums1.size() + nums2.size());
        merged.insert(merged.end(), nums1.begin(), nums1.end());
        merged.insert(merged.end(), nums2.begin(), nums2.end());

        // Step 2: Sort the merged array
        sort(merged.begin(), merged.end());

        // Step 3: Find median
        int n = merged.size();
        if (n % 2 == 1) {
            // Odd length → middle element
            return merged[n / 2];
        } else {
            // Even length → average of two middle elements
            return (merged[n / 2 - 1] + merged[n / 2]) / 2.0;
        }
    }
};

File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.java
import java.util.*;

class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {

        // Step 1: Merge both arrays
        int[] merged = new int[nums1.length + nums2.length];

        int index = 0;
        for (int num : nums1) {
            merged[index++] = num;
        }

        for (int num : nums2) {
            merged[index++] = num;
        }

        // Step 2: Sort merged array
        Arrays.sort(merged);

        // Step 3: Find median
        int n = merged.length;

        if (n % 2 == 1) {
            return merged[n / 2];
        } else {
            return (merged[n / 2 - 1] + merged[n / 2]) / 2.0;
        }
    }
}